Sets sending of analytics to cloud dashboard as default behavior

Primecutz opened the pull request:
### Short description 📝

> Describe here the purpose of your PR.

Default option for cloud configuration option was previously empty. This PR sets the default option to automatically send analytical data to the cloud dashboard. The `.analytics` option will be deprecated and a new `.disableAnalytics` can be used to disable this feature.

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/testing-strategy) for reference).

### Contributor checklist ✅

- [ ] The code has been linted using run `./fourier lint tuist --fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [ ] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Contributors have checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

danyf90 reviewed (changes requested):
This is a breaking change, so we can't do it unless we do a major release, which we don't want to do now

fortmarek commented:
> This is a breaking change, so we can't do it unless we do a major release, which we don't want to do now

This is a breaking change only for users having a completely custom Cloud backend as if you have a custom cloud backend, you might start getting requests you can't handle. While we do support custom Cloud backend, the API is not documented and thus I think we should be more benevolent here and allow changing the default behaviour.

As for the implementation itself, I think we should instead have analytics enabled by default and we can add a new option `.disableAnalytics`. `.analytics` can be deprecated and removed in the next major version.

We'd then only skip sending analytics [here](https://github.com/tuist/tuist/blob/5f2e61aead229f38f25db9ad38bea997923ee74d/Sources/TuistAnalytics/Backends/TuistAnalyticsCloudBackend.swift#L32) if `.disableAnalytics` would be present.

Primecutz commented:
Thanks for the feedback guys! I got confused for a second if it was ok to enable `analytics` by default. I think the consensus is that it is ok because tuist will not fail. Only have some errors show up if the custom backend does not support analytics?

I have updated to deprecate `analytics` and added the option to disable.

fortmarek commented:
> Only have some errors show up if the custom backend does not support analytics?

Yeah, that's right 👍

fortmarek commented on Sources/ProjectDescription/Cloud.swift:
```diff
@@ -4,13 +4,19 @@ import Foundation
+        @available(*, deprecated, message: "Enabling analytics will soon be default. Use `disableAnalytics` to disable this feature.")
         case analytics
+        
+        /// Disables sending analytics to cloud dashboard.
```
```suggestion
        /// Disables sending analytics to the cloud backend.
```

fortmarek commented on Sources/ProjectDescription/Cloud.swift:
```diff
@@ -25,7 +31,7 @@ public struct Cloud: Codable, Equatable {
     ///   - options: Cloud options.
     /// - Returns: A Cloud instance.
-    public static func cloud(projectId: String, url: String, options: [Option] = []) -> Cloud {
+    public static func cloud(projectId: String, url: String, options: [Option] = [.analytics]) -> Cloud {
```
```suggestion
    public static func cloud(projectId: String, url: String, options: [Option] = []) -> Cloud {
```

fortmarek reviewed (commented):
Just some minor comments, but this looks good!

Primecutz commented:
> Just some minor comments, but this looks good!

Thanks. Updated!

danyf90 commented:
@Primecutz can you please run the linter (`./fourier lint tuist --fix`)? 🙏

fortmarek commented on Sources/TuistAnalytics/Backends/TuistAnalyticsCloudBackend.swift:
```diff
@@ -29,7 +29,7 @@ class TuistAnalyticsCloudBackend: TuistAnalyticsBackend {
 
     func send(commandEvent: CommandEvent) async throws {
-        guard config.options.contains(.analytics) else { return }
+        guard !config.options.contains(.disableAnalytics) else { return }
```
You probably also want to edit the `TuistAnalyticsCloudBackendTests` 🙏

Primecutz replied:
Thank you. Updated.

Primecutz commented:
> @Primecutz can you please run the linter (`./fourier lint tuist --fix`)? 🙏

Thanks! Ran and updated.

fortmarek commented:
Thanks again for this PR 👏
